lambdas/scripts/dynamodb_migration_prmp_198.py

```diff
@@ -0,0 +1,160 @@
+import importlib
+import logging
+import sys
+
+from services.base.dynamo_service import DynamoDBService
+
+
+class VersionMigration:
+    def __init__(self, environment: str, table_name: str, dry_run: bool = False):
+        self.environment = environment
+        self.table_name = table_name
+        self.dynamo_service = DynamoDBService()
+        self.dry_run = dry_run
+        self.logger = logging.getLogger("VersionMigration")
+
+        self.target_table = f"{self.environment}_{self.table_name}"
+        self.bulk_upload_table = f"{self.environment}_BulkUploadReport"
+
+    def main(self):
+        self.logger.info("Starting version migration")
+        self.logger.info(f"Target table: {self.target_table}")
+        self.logger.info(f"Dry run mode: {self.dry_run}")
+
+        try:
+            all_entries = self.dynamo_service.scan_whole_table(
+                table_name=self.target_table
+            )
+            total_count = len(all_entries)
```

Comment:
Will this handle a large dataset?

Be useful to run this on `ndr-test`

Reply:
@robg-test when this is actually use it will be scanning a segment I think this is just here untill the infrastructure is complete